FEXCore/include/FEXCore/Core/Context.h

```diff
@@ -248,17 +248,24 @@ namespace FEXCore::Context {
        *
        * @param InitialRIP The starting RIP of this thread
        * @param StackPointer The starting RSP of this thread
+       * @param WhoManages The flag to determine what manages ownership of the InternalThreadState object
        * @param NewThreadState The thread state to inherit from if not nullptr.
        * @param ParentTID The thread ID that the parent is inheriting from
        *
        * @return A new InternalThreadState object for using with a new guest thread.
        */
-      FEX_DEFAULT_VISIBILITY virtual FEXCore::Core::InternalThreadState* CreateThread(uint64_t InitialRIP, uint64_t StackPointer, FEXCore::Core::CPUState *NewThreadState = nullptr, uint64_t ParentTID = 0) = 0;
+
+      // TODO: This is a temporary construct and will be removed once the frontend has full ownership of InternalThreadState objects.
+      enum class ManagedBy {
```

Comment:
Sounds like a call for add-as-deprecated?

```suggestion
      // TODO: This is a temporary construct and will be removed once the frontend has full ownership of InternalThreadState objects.
      enum class [[deprecated]] ManagedBy {
```